src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -667,6 +693,62 @@ QgsPointXY QgsMeshLayer::snapOnFace( const QgsPointXY &point, double searchRadiu
   return centroidPosition;
 }
 
+void QgsMeshLayer::resetDatasetGroupTreeItem()
+{
+  mDatasetGroupTreeRootItem.reset( new QgsMeshDatasetGroupTreeItem );
+  QList<QgsMeshDatasetGroupMetadata> metadataList;
+  for ( int i = 0; i < mDataProvider->datasetGroupCount(); ++i )
+    metadataList.append( mDataProvider->datasetGroupMetadata( i ) );
+  QgsMeshLayerUtils::createDatasetGroupTreeItems( metadataList, mDatasetGroupTreeRootItem.get(), 0 );
+  controlActiveDatasetGroupWithDisabledGroup();
+}
+
+void QgsMeshLayer::controlActiveDatasetGroupWithDisabledGroup()
```

Comment:
I had a bit of difficulty to understand what this function does - can we rename it to e.g. updateActiveDatasetGroups() ?

Reply:
and me, I have a bit difficulties to find good naming... yours seems better.